dv/cosim/spike_cosim.cc

```diff
@@ -532,6 +545,24 @@ void SpikeCosim::fixup_csr(int csr_num, uint32_t csr_val) {
       processor->put_csr(csr_num, new_val);
 #endif
       break;
+    }
+    case CSR_MCAUSE: {
+      uint32_t any_interrupt = csr_val & 0x80000000;
+      uint32_t int_interrupt = csr_val & 0x40000000;
+
+      reg_t new_val = (csr_val & 0x0000001f) | any_interrupt;
+
+      if (any_interrupt && int_interrupt) {
+        new_val |= 0x7fffffe0;
+      }
+
+#ifdef OLD_SPIKE
+      processor->set_csr(csr_num, new_val);
+#else
+      processor->put_csr(csr_num, new_val);
+#endif
+      break;
```

Comment:
This is the behavior documented here: https://ibex-core.readthedocs.io/en/latest/03_reference/exception_interrupts.html#internal-interrupts